packages/fuels-programs/src/contract/regular.rs

```diff
@@ -51,10 +51,11 @@ mod code_types {
             }
         }
 
-        pub(crate) fn code(&self) -> Vec<u8> {
+        pub(crate) fn code(&self) -> Result<Vec<u8>> {
```

Comment:
Wdyt about having `with_code` and `with_configurables` return a `Result`, trying to apply the old configurables to the new code or the new configurables to the old code. 

If that passes then we can `.expect` on all the getters and not have `Result` everywhere. 

Nice consequence is that it will fail closer to the thing that made it fail -- ie the code or the configurables that were invalid, not later on when you try and calculate the contract id.

Reply:
I like the idea.